tests/unit/customizations/eks/test_get_token.py

```diff
@@ -103,3 +114,33 @@ def test_token_sess(self):
             token_no_prefix.encode()
         ).decode()
         self.assert_url_correct(decrypted_token, True)
+
+class TestGetTokenCommand(unittest.TestCase):
+
+    @patch('awscli.customizations.eks.get_token.uni_print')
+    @patch.object(TokenGenerator, 'get_token', return_value='k8s-aws-v1.aHR0cHM6Ly9zdHMuYW1hem9uYXdzLmNvbS8=')
```

Comment:
Rather than patching the token generator, you should inject a mock that gives you the behavior you want. I think the problem here is that the `TokenGenerator` class takes region as an argument to `__init__` instead of `get_token` which means you can't accept the `TokenGenerator` as an optional argument to construct `GetTokenCommand`. If you do a small refactor to fix that, then it will be easy to inject a very simple mock in this test.

Reply:
Done